### PR TITLE
Refactor/role system

### PR DESCRIPTION
**Pull request by Deniz1923**
PR body:
  ## Summary

  - `APPLICANT` and `PROJECT_OWNER` roles removed. The system now has only `USER` and `ADMIN`.
  - Project creation is no longer gated by a role — all authenticated users can create projects.
    Ownership is already enforced per-handler (`project.owner.id == authenticatedUserId`).
  - `user_roles` join table replaces the single `users.role` column.

  ## What changes

  | Before | After |
  |--------|-------|
  | New users default to `APPLICANT` | New users default to `USER` |
  | Only `PROJECT_OWNER` could create a project | Any authenticated user can create a project |
  | Admin could promote to `PROJECT_OWNER` | No promotion needed — role has no project meaning |
  | `CHANGE_ROLE_TO_PROJECT_OWNER` bulk action | `PROMOTE_TO_ADMIN` / `DEMOTE_TO_USER` |

  ## Production deployment notes

  **Not needed in development.** For production:

  1. Deploy the new build — Hibernate creates `user_roles` automatically
  2. Run `migrate-roles.sql` immediately after to populate roles from existing data
  3. Admins are locked out of `/api/v1/admin/*` only during the window between steps 1 and 2

  Existing JWT tokens with `APPLICANT`/`PROJECT_OWNER` claims are safe — no code checks those roles anymore. They self-heal on the user's next token refresh.

  ## Test plan
  - [x] `./gradlew test` passes locally
  - [ ] Authenticated user (USER role) can create a project
  - [ ] Admin endpoints still require ADMIN role
  - [ ] Token refresh issues correct USER/ADMIN role
  - [ ] CSV import works with USER/ADMIN role values

### CODE REVIEW

File: src/main/java/com/iyte_yazilim/proje_pazari/application/commands/importUsersFromCsv/ImportUsersFromCsvHandler.java [hunk 1]
```diff
@@ -98,7 +98,7 @@ public ApiResponse<ImportResultDTO> handle(ImportUsersFromCsvCommand command) {
                     user.setEmail(email);
                     user.setFirstName(firstName);
                     user.setLastName(lastName);
-                    user.setRole(role);
+                    user.setRoles(new java.util.HashSet<>(java.util.Set.of(role)));
```
Comment (uncorrupt0) [inline src/main/java/com/iyte_yazilim/proje_pazari/application/commands/importUsersFromCsv/ImportUsersFromCsvHandler.java]:
```java
// Before
user.setRoles(new java.util.HashSet<>(java.util.Set.of(role)));

// After — add imports at top:
// import java.util.HashSet;
// import java.util.Set;
user.setRoles(new HashSet<>(Set.of(role)));
```

File: src/main/java/com/iyte_yazilim/proje_pazari/application/commands/bulkUserAction/BulkUserActionHandler.java [hunk 2]
```diff
@@ -46,8 +46,16 @@ public ApiResponse<BulkActionResult> handle(BulkUserActionCommand command) {
...
+                    case "PROMOTE_TO_ADMIN":
+                        user.getRoles().add(RoleType.ADMIN);
+                        userRepository.save(user);
+                        result.incrementSuccess();
+                        break;
+                    case "DEMOTE_TO_USER":
+                        user.getRoles().remove(RoleType.ADMIN);
+                        if (user.getRoles().isEmpty()) {
+                            user.getRoles().add(RoleType.USER);
+                        }
```
Comment (uncorrupt0) [inline src/main/java/com/iyte_yazilim/proje_pazari/application/commands/bulkUserAction/BulkUserActionHandler.java]:
```java
// Current PROMOTE_TO_ADMIN — leaves {USER, ADMIN} in the set
case "PROMOTE_TO_ADMIN":
    user.getRoles().add(RoleType.ADMIN);
    ...

// Suggested — consistent with AdminUpdateUserHandler
case "PROMOTE_TO_ADMIN":
    user.getRoles().clear();
    user.getRoles().add(RoleType.ADMIN);
    ...

// Suggested DEMOTE_TO_USER — explicit, no conditional needed
case "DEMOTE_TO_USER":
    user.getRoles().clear();
    user.getRoles().add(RoleType.USER);
    ...
```

**Review by uncorrupt0 — CHANGES_REQUESTED** [review]
# Code Review — PR #87: Refactor Role System (USER/ADMIN)

**Branch:** `refactor/role-system` → `dev`
**Author:** @Deniz1923
**CI:** All checks pass ✓ (Code Quality, Build & Test, PR Validation)
**Mergeable:** Yes (no conflicts)

---

## Summary

This PR correctly removes the `APPLICANT` and `PROJECT_OWNER` roles and replaces the single-column role model with a `user_roles` join table backed by `@ElementCollection`. The core refactor is sound and the migration script is well-crafted. A few issues need attention before merge.

---

## Issues

### Medium

**1. New bulk role actions have no test coverage**
`BulkUserActionHandlerTest.java` — no tests added for `PROMOTE_TO_ADMIN` or `DEMOTE_TO_USER`.

The old `CHANGE_ROLE_TO_PROJECT_OWNER` case was covered by unit tests (now deleted). The two new cases replace it but are completely untested. Both cases involve mutation of a `Set` under `@ElementCollection`, which is exactly the kind of logic worth exercising — especially the `DEMOTE_TO_USER` safety net (`if (roles.isEmpty()) add USER`).

Required tests:
- `PROMOTE_TO_ADMIN` → adds ADMIN to set, saves user
- `PROMOTE_TO_ADMIN` when user already has ADMIN → idempotent, still succeeds
- `DEMOTE_TO_USER` → removes ADMIN, USER remains
- `DEMOTE_TO_USER` on a user with only USER role → no-op, USER still present

---

**2. `PROMOTE_TO_ADMIN` leaves USER and ADMIN coexisting in the role set**
`BulkUserActionHandler.java:80`

```java
// BulkUserActionHandler — just adds ADMIN
user.getRoles().add(RoleType.ADMIN);
```

```java
// AdminUpdateUserHandler — replaces all roles
user.getRoles().clear();
user.getRoles().add(command.role());
```

After `PROMOTE_TO_ADMIN`, a user ends up with `{USER, ADMIN}`. After `AdminUpdateUserHandler` with `ADMIN`, the user has `{ADMIN}` only. The JWT logic (`contains(ADMIN) ? "ADMIN" : "USER"`) handles both correctly, but the database state is inconsistent between the two promotion paths.

Recommendation: Use `clear() + add(ADMIN)` in `PROMOTE_TO_ADMIN` as well, consistent with `AdminUpdateUserHandler`. For `DEMOTE_TO_USER`, always do `clear() + add(USER)`.

---

**3. Export/import roundtrip is broken for multi-role users**
`ExportUsersHandler.java:271-276` and `ImportUsersFromCsvHandler.java:84`

The export now renders multiple roles as pipe-delimited: `USER|ADMIN`.

```java
// ExportUsersHandler — pipe-delimited output
.reduce((a, b) -> a + "|" + b)
```

But the import parser calls `RoleType.valueOf(roleStr.toUpperCase())` on the raw field value. `USER|ADMIN` is not a valid `RoleType` name and throws `IllegalArgumentException`, which the handler catches and records as a row-level error.

Since the system currently only has USER and ADMIN, the immediate risk is low (a user would need both roles simultaneously, which only happens via `PROMOTE_TO_ADMIN` as noted above). However, fixing issue #2 would eliminate the dual-role state entirely and make this moot. If dual roles are intentional, the import handler must be updated to split on `|` before parsing.

---

### Low

**4. Inline fully-qualified class names in `ImportUsersFromCsvHandler`**
`ImportUsersFromCsvHandler.java:101`

```java
user.setRoles(new java.util.HashSet<>(java.util.Set.of(role)));
```

`HashSet` and `Set` are not imported at the top of this file. All other modified files in this PR use proper import statements. This will cause `spotlessCheck` to flag a style inconsistency on any re-format. Replace with a proper import and a clean expression.

---

**5. PR test plan has unchecked items**

The PR body shows four unchecked manual test items:
- `[ ]` Authenticated user (USER role) can create a project
- `[ ]` Admin endpoints still require ADMIN role
- `[ ]` Token refresh issues correct USER/ADMIN role
- `[ ]` CSV import works with USER/ADMIN role values

These are straightforward to verify with the existing integration tests or a quick local smoke test. Please check them off before marking ready.

---

**6. PR title is not conventional-commit style**

Current: `Refactor/role system` (auto-generated from branch name).
Suggested: `refactor(auth): simplify role system to USER/ADMIN`

---

**7. No linked issue in PR body**

The PR does not contain `Closes #N`. If this work was tracked under an issue, add the link. If not, this is fine — just noting it for completeness.

---

## Positive Observations

- The migration script is well-written: idempotent, transactional, INSERT-only, with rollback notes and a verification query. Excellent production-safety approach.
- `DISTINCT` correctly added to `findWithFilters` query to prevent row duplication from the `LEFT JOIN` on roles.
- `UserMapper` now correctly maps roles bidirectionally; the old `@Mapping(target = "roles", ignore = true)` comment-hacks are gone.
- `LoginUserHandler` and `RefreshTokenHandler` role determination (`contains(ADMIN) ? "ADMIN" : "USER"`) is clean and handles the multi-role case correctly.
- `DEMOTE_TO_USER` correctly guards against leaving a user with no roles.
- `@ElementCollection(fetch = FetchType.EAGER)` is the right fetch strategy here — roles must be available in the security context without lazy-load exceptions.
- Removal of `PromoteToProjectOwnerCommand/Handler` is clean with full test cleanup.

**Comment by Deniz1923** [comment]
▎ Addressed all medium and low issues from the review:

  ▎ #2 — PROMOTE_TO_ADMIN role set inconsistency
  ▎ Both PROMOTE_TO_ADMIN and DEMOTE_TO_USER now use clear() + add(...), matching the pattern in
  AdminUpdateUserHandler. Users will always carry exactly one role after either operation.

  ▎ #3 — Export/import roundtrip
  ▎ No longer applicable — fixing #2 eliminates the dual-role state that caused USER|ADMIN to appear in
  exports.

  ▎ #1 — Missing test coverage
  ▎ Added four tests to BulkUserActionHandlerTest: promote (USER→ADMIN), idempotent promote
  (ADMIN→ADMIN), demote (ADMIN→USER), and demote no-op (USER→USER).

  ▎ #4 — Inline FQCNs in ImportUsersFromCsvHandler
  ▎ Added HashSet and Set imports; replaced the inline qualified names on line 101.

File: src/main/java/com/iyte_yazilim/proje_pazari/application/commands/loginUser/LoginUserHandler.java [hunk 3]
```diff
@@ -79,7 +79,10 @@ public ApiResponse<LoginUserResult> handle(LoginUserCommand command) {
         }
 
         // --- 6. Generate JWT token with userId, email, and role ---
-        String role = user.getRole() != null ? user.getRole().toString() : "APPLICANT";
+        String role =
+                user.getRoles().contains(com.iyte_yazilim.proje_pazari.domain.enums.RoleType.ADMIN)
```
Comment (uncorrupt0) [inline src/main/java/com/iyte_yazilim/proje_pazari/application/commands/loginUser/LoginUserHandler.java]: Add `import com.iyte_yazilim.proje_pazari.domain.enums.RoleType;` and use `RoleType.ADMIN` — same fix applied to `ImportUsersFromCsvHandler` in this PR.

File: src/main/java/com/iyte_yazilim/proje_pazari/application/commands/refreshToken/RefreshTokenHandler.java [hunk 4]
```diff
@@ -42,7 +42,10 @@ public ApiResponse<RefreshTokenResult> handle(RefreshTokenCommand command) {
 
         refreshTokenService.revokeRefreshToken(refreshToken);
         String newRefreshToken = refreshTokenService.createRefreshToken(userId);
-        String role = user.getRole() != null ? user.getRole().toString() : "APPLICANT";
+        String role =
+                user.getRoles().contains(com.iyte_yazilim.proje_pazari.domain.enums.RoleType.ADMIN)
```
Comment (uncorrupt0) [inline src/main/java/com/iyte_yazilim/proje_pazari/application/commands/refreshToken/RefreshTokenHandler.java]: Same inline FQCN as `LoginUserHandler:83`. `RoleType` is not imported.

File: src/main/java/com/iyte_yazilim/proje_pazari/application/queries/exportUsers/ExportUsersHandler.java [hunk 5]
```diff
@@ -30,7 +30,13 @@ public ApiResponse<String> handle(ExportUsersQuery query) {
                     .append(",")
                     .append(escapeCsv(user.getLastName()))
                     .append(",")
-                    .append(user.getRole() != null ? user.getRole().name() : "")
+                    .append(
+                            user.getRoles() != null
```
Comment (uncorrupt0) [inline src/main/java/com/iyte_yazilim/proje_pazari/application/queries/exportUsers/ExportUsersHandler.java]: `getRoles()` on `@ElementCollection` is never null. Drop the null guard.

File: src/test/java/com/iyte_yazilim/proje_pazari/application/commands/adminUpdateUser/AdminUpdateUserHandlerTest.java [hunk 6]
```diff
@@ -43,12 +45,12 @@ void shouldUpdateUserRole() {
...
         AdminUpdateUserCommand command =
                 new AdminUpdateUserCommand(
-                        "01ABCDEF12345678901234", RoleType.PROJECT_OWNER, null, null, null, null);
+                        "01ABCDEF12345678901234", RoleType.ADMIN, null, null, null, null);
 
         ApiResponse<Void> response = handler.handle(command);
 
         assertNotNull(response);
-        assertEquals(RoleType.PROJECT_OWNER, testUser.getRole());
+        assertTrue(testUser.getRoles().contains(RoleType.ADMIN));
```
Comment (uncorrupt0) [inline src/test/java/com/iyte_yazilim/proje_pazari/application/commands/adminUpdateUser/AdminUpdateUserHandlerTest.java]: Weak assertion — use `assertEquals(Set.of(RoleType.ADMIN), testUser.getRoles())` to also verify `clear()` removed `USER`.

**Review by uncorrupt0 — CHANGES_REQUESTED** [review]
# Code Review — PR #87 Follow-up: Refactor Role System (USER/ADMIN)

**Branch:** `refactor/role-system` → `dev`
**Author:** @Deniz1923
**CI:** All checks pass ✓ (Code Quality, Build & Test, PR Validation)
**Reviewing commit:** 66c8bba (latest, post-fix push)
**Previous review:** @uncorrupt0 — `CHANGES_REQUESTED`

---

## Status of Previous Review Items

| # | Issue | Status |
|---|-------|--------|
| 1 | No test coverage for `PROMOTE_TO_ADMIN` / `DEMOTE_TO_USER` | ✅ Fixed — 4 tests added |
| 2 | `PROMOTE_TO_ADMIN` leaves `{USER, ADMIN}` in role set | ✅ Fixed — now `clear() + add()` |
| 3 | Export/import roundtrip broken for multi-role users | ✅ Resolved by fixing #2 |
| 4 | Inline FQCNs in `ImportUsersFromCsvHandler` | ✅ Fixed — `HashSet`/`Set` imported |
| 5 | Unchecked test plan items | ⚠️ Still unchecked in PR body |
| 6 | PR title not conventional-commit style | ⚠️ Still "Refactor/role system" |
| 7 | No linked issue (`Closes #N`) | ⚠️ Still missing (if applicable) |

The core logic fixes are solid. Issues #1–#4 are resolved. Three hygiene items
(#5–#7) remain open from the previous review, and this pass found two new issues
the previous review missed.

---

## New Issues

### Medium

**1. Inline FQCNs in `LoginUserHandler` and `RefreshTokenHandler` — missed by previous review**

The previous review correctly flagged `ImportUsersFromCsvHandler` for inline
fully-qualified class names (issue #4) and it was fixed. The same pattern exists in
two other files modified by this PR that were not caught at the time.

`LoginUserHandler.java:83`:
```java
user.getRoles().contains(com.iyte_yazilim.proje_pazari.domain.enums.RoleType.ADMIN)
```

`RefreshTokenHandler.java:46`:
```java
user.getRoles().contains(com.iyte_yazilim.proje_pazari.domain.enums.RoleType.ADMIN)
```

`RoleType` is not imported in either file. Fix: add
`import com.iyte_yazilim.proje_pazari.domain.enums.RoleType;` to both and replace
the FQCN with `RoleType.ADMIN`. This is the same fix already applied to
`ImportUsersFromCsvHandler` in this PR.

---

### Low

**2. Dead null check in `ExportUsersHandler`**

`ExportUsersHandler.java:34`:
```java
user.getRoles() != null
    ? user.getRoles().stream().map(Enum::name).reduce((a, b) -> a + "|" + b).orElse("")
    : ""
```

`roles` is declared as `private Set<RoleType> roles = new HashSet<>()` in
`UserEntity`, and `@ElementCollection` fields are managed by Hibernate as
`PersistentSet` — they are never null (an empty collection, not null, when there
are no rows). The `!= null` guard is unreachable dead code.

Simplify to:
```java
user.getRoles().stream()
    .map(Enum::name)
    .reduce((a, b) -> a + "|" + b)
    .orElse("")
```

---

**3. `AdminUpdateUserHandlerTest` does not assert `clear()` behavior**

`AdminUpdateUserHandlerTest.java:53`:
```java
assertTrue(testUser.getRoles().contains(RoleType.ADMIN));
```

The test starts with `testUser.roles = {USER}` and calls `AdminUpdateUserHandler`
with role `ADMIN`. The handler does `clear() + add(ADMIN)`, so the post-condition
should be `{ADMIN}` only. The current assertion only checks that ADMIN was added —
it does not verify USER was removed. If someone reverts to `add()` without `clear()`
in the future, this test still passes.

The `BulkUserActionHandlerTest` added in this PR uses the stronger pattern:
```java
assertEquals(Set.of(RoleType.ADMIN), testUser.getRoles());
```
Apply the same here.

---

## Open Items From Previous Review

**#5 — Test plan items still unchecked**

Four manual verification items remain unchecked in the PR body. Please verify and
check them off before marking ready for merge.

**#6 — PR title**

Still auto-generated: "Refactor/role system". Suggested:
`refactor(auth): simplify role system to USER/ADMIN`

**#7 — No linked issue**

No `Closes #N` in the PR body. Add if this work was tracked under an issue.

---

## Positive Observations

All four new `BulkUserActionHandlerTest` cases are well-structured and cover exactly
the edge cases from the previous review (promote, idempotent promote, demote,
demote no-op). The `assertEquals(Set.of(...))` assertion pattern used there is
stronger than a `contains()` check and correctly validates both add and clear
behavior simultaneously. Set that as the bar for the remaining test fix (#3 above).

---

## Recommendation

**Request changes** — issues #1 and #3 are straightforward fixes that should go in
before merge. Issue #2 is a cleanup. Issues #5–#7 are process items to resolve
before `gh pr ready`.

**Comment by Deniz1923** [comment]
▎ Addressed all issues from the second review:

  ▎ #1 — Inline FQCNs in LoginUserHandler and RefreshTokenHandler
  ▎ Added `import RoleType` to both files and replaced the fully-qualified
  `com.iyte_yazilim.proje_pazari.domain.enums.RoleType.ADMIN` with `RoleType.ADMIN`.

  ▎ #2 — Dead null check in ExportUsersHandler
  ▎ Removed the unreachable `user.getRoles() != null` guard — `@ElementCollection` fields
  are managed by Hibernate as `PersistentSet` and are never null.

  ▎ #3 — Weak assertion in AdminUpdateUserHandlerTest
  ▎ Replaced `assertTrue(contains(RoleType.ADMIN))` with
  `assertEquals(Set.of(RoleType.ADMIN), testUser.getRoles())` to verify that `clear()`
  removed USER. Matches the stronger pattern already used in BulkUserActionHandlerTest.

  ▎ Additional fix — Inline FQCN in UserController
  ▎ Found and fixed `new java.util.HashMap<>()` on line 209 — added `HashMap` import,
  same pattern as the reviewer flagged elsewhere.

  ▎ `spotlessApply` and `./gradlew test` both pass.

File: src/main/java/com/iyte_yazilim/proje_pazari/infrastructure/persistence/models/UserEntity.java [hunk 7]
```diff
@@ -89,8 +97,9 @@ protected void onCreate() {
         if (isActive == null) {
             isActive = true;
         }
-        if (role == null) {
-            role = RoleType.APPLICANT;
+        if (roles == null || roles.isEmpty()) {
+            roles = new HashSet<>();
+            roles.add(RoleType.USER);
```
Comment (uncorrupt0) [inline src/main/java/com/iyte_yazilim/proje_pazari/infrastructure/persistence/models/UserEntity.java:102]: The `roles == null` branch is unreachable — the field is initialized to `new HashSet<>()` at declaration. The `isEmpty()` check is the meaningful guard. This is a minor readability nit, not a bug, and is consistent with how the old `role == null` guard was written. No action required; noting for awareness.

File: src/main/java/com/iyte_yazilim/proje_pazari/application/queries/exportUsers/ExportUsersHandler.java [hunk 8]
```diff
@@ -30,7 +30,11 @@ public ApiResponse<String> handle(ExportUsersQuery query) {
                     .append(",")
                     .append(escapeCsv(user.getLastName()))
                     .append(",")
-                    .append(user.getRole() != null ? user.getRole().name() : "")
+                    .append(
+                            user.getRoles().stream()
+                                    .map(Enum::name)
+                                    .reduce((a, b) -> a + "|" + b)
+                                    .orElse(""))
```
Comment (uncorrupt0) [inline src/main/java/com/iyte_yazilim/proje_pazari/application/queries/exportUsers/ExportUsersHandler.java]: The set iteration order of `HashSet` is not guaranteed. If a user ever has two roles (e.g., `{USER, ADMIN}`), the exported string could be `USER|ADMIN` or `ADMIN|USER` depending on JVM hash state. Since the import parser does not handle pipe-delimited values anyway (and dual-role state cannot occur given the `clear() + add()` pattern), this is not a live issue. Worth a comment to that effect if the export format is ever documented or the import is updated.

**Review by uncorrupt0 — APPROVED** [review]
# Code Review — PR #87 Third Pass: Refactor Role System (USER/ADMIN)

**Branch:** `refactor/role-system` → `dev`
**Author:** @Deniz1923
**CI:** All checks pass ✓ (Code Quality, Build & Test, PR Validation)
**Reviewing commit:** 5bc1612 (latest)
**Previous reviews:** @uncorrupt0 — `CHANGES_REQUESTED` × 2

---

## Status of All Previous Review Items

| # | Review | Issue | Status |
|---|--------|-------|--------|
| 1 | First | No test coverage for `PROMOTE_TO_ADMIN` / `DEMOTE_TO_USER` | ✅ Fixed |
| 2 | First | `PROMOTE_TO_ADMIN` leaves `{USER, ADMIN}` in role set | ✅ Fixed |
| 3 | First | Export/import roundtrip broken for multi-role users | ✅ Resolved by fixing #2 |
| 4 | First | Inline FQCNs in `ImportUsersFromCsvHandler` | ✅ Fixed |
| 5 | First | Unchecked test plan items | ⚠️ Still open |
| 6 | First | PR title not conventional-commit style | ⚠️ Still open |
| 7 | First | No linked issue (`Closes #N`) | ⚠️ Still open |
| 8 | Second | Inline FQCNs in `LoginUserHandler` / `RefreshTokenHandler` | ✅ Fixed |
| 9 | Second | Dead null check in `ExportUsersHandler` | ✅ Fixed |
| 10 | Second | Weak assertion in `AdminUpdateUserHandlerTest` | ✅ Fixed |

All code-level issues from both previous reviews are resolved. Three process items (#5, #6, #7) remain open and are addressed below.

---

## Verification of Latest Fixes

**#8 — Inline FQCNs** (`LoginUserHandler.java`, `RefreshTokenHandler.java`)

Both files now have `import com.iyte_yazilim.proje_pazari.domain.enums.RoleType;` at the top and reference `RoleType.ADMIN` directly. Clean.

**#9 — Dead null check** (`ExportUsersHandler.java`)

The null guard is gone. The expression is now:

```java
user.getRoles().stream()
    .map(Enum::name)
    .reduce((a, b) -> a + "|" + b)
    .orElse("")
```

Correct — `roles` is initialized to `new HashSet<>()` and is never null.

**#10 — Weak assertion** (`AdminUpdateUserHandlerTest.java:53`)

```java
// Before
assertTrue(testUser.getRoles().contains(RoleType.ADMIN));

// After
assertEquals(Set.of(RoleType.ADMIN), testUser.getRoles());
```

Correct — the stronger assertion now validates both that ADMIN was added and that USER was cleared. Matches the pattern already used in `BulkUserActionHandlerTest`.

---

## Open Process Items

**#5 — Test plan items still unchecked**

The PR body still shows four unchecked manual verification items. These should be checked off before converting to ready for review. They are straightforward to verify against the existing integration tests or a local smoke test.

**#6 — PR title**

Still auto-generated from the branch name: `Refactor/role system`.
Suggested: `refactor(auth): simplify role system to USER/ADMIN`

Update via: `gh pr edit 87 --title "refactor(auth): simplify role system to USER/ADMIN"`

**#7 — No linked issue**

Still no `Closes #N` in the PR body. If this work is tracked under an issue, add the link. If not, confirm in the PR body that no issue exists for traceability.

---

## Recommendation

**Approve** — all code issues are resolved. The implementation is correct and the test coverage is solid. Issues #5–#7 are housekeeping: complete the manual test plan verification, update the PR title to follow conventional-commit style, and add a `Closes #N` link if applicable, then `gh pr ready`.

---

## Positive Observations

The full refactor is well-executed:

- `clear() + add()` pattern used consistently across `BulkUserActionHandler`, `AdminUpdateUserHandler`, and the new tests — role set state is always deterministic.
- `BulkUserActionHandlerTest` covers all four edge cases (promote, idempotent promote, demote, demote no-op) with strong `assertEquals(Set.of(...))` assertions.
- `AdminUpdateUserHandlerTest` assertion upgraded to match the stronger pattern from the bulk tests.
- `ExportUsersHandler` is now clean — dead null guard removed, stream pipeline is direct.
- `LoginUserHandler` and `RefreshTokenHandler` both use proper imports; no inline FQCNs remain anywhere in the PR.
- Migration script remains production-safe: idempotent, INSERT-only, transactional, with rollback and verification instructions.
